### ISSUE TITLE
AGIC unable to create listeners with hostnames

### ISSUE THREAD

**Issue by vishal8k** [Azure/application-gateway-kubernetes-ingress#941]
**Describe the bug**
Below is my ingress yaml file. 
```
kind: Ingress
apiVersion: extensions/v1beta1
metadata:
  name: master-ingress
  namespace: mynamespace
  annotations:
    kubernetes.io/ingress.class: azure/application-gateway
    appgw.ingress.kubernetes.io/use-private-ip: "true"
spec:
  rules:
  - host: api.example.com
    http:
      paths:
      - backend:
          serviceName: first-service
          servicePort: 80
        path: /
  - host: sub.example.com
    http:
      paths:
      - backend:
          serviceName: second-service
          servicePort: 80
        path: /
```
I have custom resource definition to prohibit AGIC from removing manually added targets and that works fine. AGIC does not remove them.  but however the above ingress is not reflected and errors from ingress pod is listed below. 
**To Reproduce**
Normal brownfield deployment of AGIC using helm. 

**Ingress Controller details**
* Output of `kubectl describe pod <ingress controller`> . The <ingress controller> pod name can be obtained by running `helm list`. 
 I find the below error from the pod logs 

```
I0708 05:35:50.170635       1 listeners.go:103] [brownfield] Listeners AGIC created: fl-9a24270230c96caf5867b04cd957b0c9, fl-c277e124511befba0ebe69632af7079f
I0708 05:35:50.171559       1 routing_rules.go:95] [brownfield] Rules AGIC created: rr-e1903c8aa3446b7b3207aec6d6ecba8a, rr-d712c965fd0d8ae34d6363d4a1671663
Error: Code="ApplicationGatewayHttpListenersUsingSameFrontendPort" Message="Two Http Listeners of Application Gateway /subscriptions/e3d6f38e-68ab-4f1e-9ceb-18373bc9ea68/resourceGroups/myrg/providers/Microsoft.Network/applicationGateways/myappgw are using the same Frontend Port /subscriptions/e3d6f38e-68ab-4f1e-9ceb-18373bc9ea68/resourceGroups/myrg/providers/Microsoft.Network/applicationGateways/myappgw/frontendPorts/fp-80." Details=[]
E0708 05:35:50.382499       1 worker.go:62] Error processing event.network.ApplicationGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="ApplicationGatewayHttpListenersUsingSameFrontendPort" Message="Two Http Listeners of Application Gateway /subscriptions/e3d6f38e-68ab-4f1e-9ceb-18373bc9ea68/resourceGroups/msd-rg-madstacklivetest-sea/providers/Microsoft.Network/applicationGateways/agw-dev-pr-mad-sa-01 are using the same Frontend Port /subscriptions/e3d6f38e-68ab-4f1e-9ceb-18373bc9ea68/resourceGroups/msd-rg-madstacklivetest-sea/providers/Microsoft.Network/applicationGateways/agw-dev-pr-mad-sa-01/frontendPorts/fp-80." Details=[]
```
The error doesn't make any sense because It can be clearly seen from my ingress yaml itself that I use the same listener but I use different hostnames. Now the same setup is working for me in another subscription. but this time it failed with a weird error.

**Comment by akshaysngupta** [comment]
@vishal8k Your ingress is annotated with `use-private-ip`. This means that the listener will use the private IP with port 80.
Do you have another rule using port 80 but on public IP? If yes, then that might be the problem.
(Currently, AppGW can't have both public ip and private ip on the same port. We are working on supporting this feature but don't have an ETA.)

**Comment by vishal8k** [comment]
@akshaysngupta I raised a support case with Azure and they too confirmed that this was due to a problem in underlying API (Azure side). They are yet to trace down a root cause but. and also to have two listeners on same port (public and private) is possible if we have multi-site. I have the setup already in my production and it is working smoothly.

**Comment by SetarehStar** [comment]
I have the same issue even not using private IP. Can some one please let me know if it is possible to have multi site listener using AGIC?

**Comment by vishal8k** [comment]
@SetarehStar As I mentioned above Azure support confirmed that the error I saw was due to the failures in APPGW background (Managed by Azure). To answer your query, It is possible to have two listeners on the same port with multi-site (Irrespective of public or private). More funnier thing is the support Agent who handled my case initially told it is not possible, Only after showing my working setup he came to know that it is possible *facepalm*. Azure is deteriorating nowadays.
